src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -107,16 +107,26 @@ private static byte HexToByte(char val)
                 return 0xFF;
         }
 
-        public static bool ContentsEqual(this byte[] a1, byte[] a2)
+        public static bool ContentsEqual(this byte[]? a1, byte[]? a2)
         {
-            if (a1.Length != a2.Length)
+            if (a1 == null)
+            {
+                return a2 == null;
+            }
+
+            if (a2 == null || a1.Length != a2.Length)
+            {
                 return false;
+            }
 
             for (int i = 0; i < a1.Length; i++)
             {
                 if (a1[i] != a2[i])
+                {
                     return false;
+                }
             }
```

Comment:
How long are these arrays generally?  Wondering if we should just make this:
```C#
return a1.AsSpan().SequenceEquals(a2);
```

Reply:
The places that triggered this change are usually small (32 bytes or less), or an RSA modulus (~2048-bit / 256 byte).